pallets/gear/src/tests.rs

```diff
@@ -2640,10 +2643,231 @@ fn test_two_contracts_composition_works() {
+// up successfully with messages sent from program with total value more than was provided to the program.
+//
+// Again init message won't be added to the queue, because of the check here (https://github.com/gear-tech/gear/blob/master/pallets/gear/src/manager.rs#L351-L364).
+// But it's is not preferable to enter that `if` clause.
+// todo # 929 After create_program sys-call becomes fallible, tests must be changed
+#[test]
+fn test_create_program_with_exceeding_value() {
+    use demo_init_with_value::{SendMessage, WASM_BINARY};
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        // Submit the code
+        assert_ok!(GearPallet::<Test>::submit_code(
+            Origin::signed(USER_1),
+            ProgramCodeKind::Default.to_bytes(),
+        ));
+
+        let sending_to_program = 2 * get_ed();
+        let random_receiver = 1;
+        // Trying to send init message from program with value greater than program can send.
+        assert_ok!(GearPallet::<Test>::submit_program(
+            Origin::signed(USER_1),
+            WASM_BINARY.to_vec(),
+            b"test1".to_vec(),
+            vec![
+                SendMessage::Handle(random_receiver, sending_to_program / 3),
+                SendMessage::Handle(random_receiver, sending_to_program / 3),
+                SendMessage::Init(sending_to_program + 1),
+            ]
+            .encode(),
+            10_000_000_000,
+            sending_to_program,
+        ));
+
+        run_to_block(2, None);
+
+        // Check there are no messages for `random_receiver`. There would be messages in mailbox
+        // if execution didn't end up with an "Not enough value to send message" error.
+        let origin_msg_id =
+            MessageId::generate_from_user(1, ProgramId::from_origin(USER_1.into_origin()), 0);
+        let receiver_mail_msg1 = MessageId::generate_outgoing(origin_msg_id, 0);
+        let receiver_mail_msg2 = MessageId::generate_outgoing(origin_msg_id, 1);
+        assert!(!Mailbox::<Test>::contains_key(
+            random_receiver,
+            receiver_mail_msg1
+        ));
+        assert!(!Mailbox::<Test>::contains_key(
+            random_receiver,
+            receiver_mail_msg2
+        ));
+
+        // User's message execution will result in trap, because program tries
+        // to send init message with value more than program has. As a result, 1 dispatch
+        // is dequeued (user's  message) and one message is sent to mailbox.
+        let mailbox_msg_id = MessageId::from_origin(get_last_message_id());
+        assert!(Mailbox::<Test>::contains_key(USER_1, mailbox_msg_id));
+        // This check means, that program's invalid init message didn't reach the queue.
+        check_dequeued(1);
+
+        // There definitely should be event with init failure reason
+        let expected_failure_reason = "Not enough value to send message".to_string().into_bytes();
```

Comment:
I guess we don't need to convert this ANSI string to `String`.
```suggestion
        let expected_failure_reason = b"Not enough value to send message"; // Or `.to_vec()`
```